src/main/java/com/exasol/adapter/dynamodb/mapping/AbstractTableMappingFactory.java

```diff
@@ -41,73 +25,140 @@ protected AbstractTableMappingFactory() {
-        final JsonColumnMappingFactory columnMappingFactory = new JsonColumnMappingFactory();
-        switch (getMappingType(definition)) {
-        case TO_STRING_MAPPING_KEY:
-            columnMappingFactory.addStringColumnIfPossible(definition.getJsonObject(TO_STRING_MAPPING_KEY), sourcePath,
-                    tableBuilder, propertyName, isRootLevel);
-            break;
-        case TO_JSON_MAPPING_KEY:
-            columnMappingFactory.addToJsonColumn(definition.getJsonObject(TO_JSON_MAPPING_KEY), sourcePath,
-                    tableBuilder, propertyName);
-            break;
-        case TO_TABLE_MAPPING_KEY:
-            queueAddingNestedTable(definition.getJsonObject(TO_TABLE_MAPPING_KEY), sourcePath, propertyName);
-            break;
-        case FIELDS_KEY:
-            visitChildren(definition.getJsonObject(FIELDS_KEY), sourcePath, tableBuilder);
-            break;
-        case "":// no mapping definition
-            break;
-        default:
-            throw new UnsupportedOperationException("This mapping type is not supported in the current version.");
-        }
-    }
+    protected abstract TableMappingDefinition createTable(final JsonObject definition,
+            List<ColumnMappingDefinition> columns);
 
-    private void queueAddingNestedTable(final JsonObject definition, final DocumentPathExpression.Builder sourcePath,
-            final String propertyName) {
-        this.nestedTableReaderQueue
-                .add(parentTable -> new NestedTableMappingFactory(parentTable, propertyName, sourcePath)
-                        .readMappingDefinition(definition));
+    protected abstract DocumentPathExpression.Builder getPathToTable();
+
+    protected abstract boolean isNestedTable();
+
+    @FunctionalInterface
+    private static interface NestedTableReader {
+        List<TableMappingDefinition> readNestedTable(TableMappingDefinition parentTable);
     }
 
-    private String getMappingType(final JsonObject definition) {
-        final Set<String> keys = definition.keySet();
-        if (keys.isEmpty()) {
-            return "";
-        } else if (keys.size() == 1) {
-            return keys.iterator().next();
-        } else {
-            throw new ExasolDocumentMappingLanguageException("Please define only one mapping for one property.");
+    private static class SchemaMappingDefinitionLanguageVisitor {
+        private static final String FIELDS_KEY = "fields";
+        private static final String TO_STRING_MAPPING_KEY = "toStringMapping";
+        private static final String TO_JSON_MAPPING_KEY = "toJsonMapping";
+        private static final String TO_TABLE_MAPPING_KEY = "toTableMapping";
+
+        /**
+         * Building tables for nested lists is delayed using this queue as they need the completely built table that
+         * maps the object they are nested in.
+         */
+        private final List<NestedTableReader> nestedTableReaderQueue;
```

Comment:
The instance name `nestedTableReaderQueue` suggests it behaves as a queue (FIFO), but the data structure is a List. 

Would it make sense to rename the variable name, or change it to Queue interface and LinkedList?

Reply:
yes ;-)